qiskit_dynamics/models/generator_models.py

```diff
@@ -50,85 +50,107 @@ class BaseGeneratorModel(ABC):
           with G'(t). Furthermore, all evaluation functions have the option
           to return the results in a basis in which :math:`F` is diagonalized,
           to save on the cost of computing :math:`e^{\pm tF}`.
-        - A `cutoff_freq`: a cutoff frequency for further modifying the
-          evaluation of :math:`G'(t)` to remove terms above a given frequency.
-          In this abstract class the exact meaning of this is left unspecified;
-          it is left to concrete implementations to define this.
     """
 
     @property
     @abstractmethod
-    def frame(self) -> BaseFrame:
-        """Get the frame."""
+    def hilbert_space_dimension(self) -> int:
```

Comment:
I would rather just call this `dim`. While it's technically a Hilbert space, that terminology is somewhat overpowered in this context.